test/specs/utils/env.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * LiskHQ/lisky
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import os from 'os';
+
+import {
+	setUpFsStubs,
+	setUpConsoleStubs,
+	setUpProcessStubs,
+} from '../../steps/utils';
+import * as given from '../../steps/1_given';
+import * as when from '../../steps/2_when';
+import * as then from '../../steps/3_then';
+
+const CONFIG_PATH = '../../../src/utils/env';
+
+describe('env util', () => {
+	beforeEach(() => {
+		setUpFsStubs();
+		setUpConsoleStubs();
+		setUpProcessStubs();
+		delete require.cache[require.resolve(CONFIG_PATH)];
+	});
+
+	describe('Given a default config', () => {
+		beforeEach(given.aDefaultConfig);
+
+		describe(`Given a directory path "${os.homedir()}/.lisky"`, () => {
+			beforeEach(given.aDirectoryPath);
+
+			describe('Given a config file name "config.json"', () => {
+				beforeEach(given.aConfigFileName);
+
+				describe('Given the directory does not exist', () => {
+					beforeEach(given.theDirectoryDoesNotExist);
+
+					describe('Given the directory cannot be created', () => {
+						beforeEach(given.theDirectoryCannotBeCreated);
+
+						describe('When the config is loaded', () => {
+							beforeEach(when.theConfigIsLoaded);
+
+							it('Then the user should be warned that the config will not be persisted', then.theUserShouldBeWarnedThatTheConfigWillNotBePersisted);
+							it('Then the default config should be exported', then.theDefaultConfigShouldBeExported);
+						});
+					});
+
+					describe('Given the config directory can be created', () => {
+						beforeEach(given.theDirectoryCanBeCreated);
```

Comment:
The specification file shouldn't need to know anything about implementation - the fact that this particular `Given` step doesn't require any code to be run is irrelevant to the structure of this file. (For a practical example, imagine if we didn't have the `beforeEach` hook here and later changed the default behaviour of stubs - perhaps we'd need to add it in just because our step implementation changed.)